### PR TITLE
NXP: Add support for the NXP Analog Comparator (ACMP) IP

### PR DESCRIPTION
**Pull request by henrikbrixandersen**
Add support for the NXP Analog Comparator (ACMP) IP present in Kinetis/i.MX RT6XX SoCs.

The driver shim was implemented as a Zephyr sensor driver and supports configuration of various parameters through devicetree or sensor attributes. The sensor driver provides a single channel for reading the output value of the analog comparator along with triggers for output rising/falling events.

The driver and sample was tested and verified on the NXP KE1xF SoC series.

### CODE REVIEW

File: west.yml [hunk 1]
```diff
@@ -102,7 +102,7 @@ manifest:
       revision: 1c4fdba512b268033a4cf926bddd323866c3261a
       path: tools/net-tools
     - name: hal_nxp
-      revision: 68bbdbdec1153355b14386d8f095058a496f66f4
+      revision: pull/58/head
```
Comment (henrikbrixandersen) [inline west.yml:105]: This PR depends on https://github.com/zephyrproject-rtos/hal_nxp/pull/58 being merged.

Comment (henrikbrixandersen) [reply west.yml]: Done.

**Comment by carlescufi** [comment]
@MaureenHelm could you take a look?

**Comment by henrikbrixandersen** [comment]
@MaureenHelm ping

**Review by MaureenHelm — APPROVED** [review]

**Comment by MaureenHelm** [comment]
@mmahadevan108 will you please have a look?

**Comment by henrikbrixandersen** [comment]
@mmahadevan108 ping?

File: samples/sensor/mcux_acmp/src/main.c [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+#include <stdio.h>
+
+#ifdef CONFIG_BOARD_TWR_KE18F
+#define ACMP_NODE  DT_NODELABEL(cmp2)
+#define ACMP_LABEL DT_LABEL(ACMP_NODE)
+#define ACMP_POSITIVE 5
+#define ACMP_NEGATIVE 5
+#define ACMP_DAC_VREF 0
+#else
+#error Unsupported board
```
Comment (mbolivar-nordic) [inline samples/sensor/mcux_acmp/src/main.c:21]: The usual thing to do would be to rely on a sample-specific alias. Why not do that here and avoid a proliferation of `#ifdef CONFIG_BOARD_...`?

Comment (henrikbrixandersen) [reply samples/sensor/mcux_acmp/src/main.c:21]: @mbolivar-nordic Thanks for reviewing! There are other, board-specific configurations needed as well (as shown in the quoted code above). I do not predict this sample to become enabled for a wide variety of boards, so this list of `#ifdefs` should not become too long.

**Review by mbolivar-nordic — APPROVED** [review]
Seems sane (er, ... seems like it's ... twistable?)

**Comment by MaureenHelm** [comment]
Will you please rebase and update for #30467?

**Comment by henrikbrixandersen** [comment]
> Will you please rebase and update for #30467?

@MaureenHelm Done, thanks for reminding me.

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**